util/chplenv/chpl_home_utils.py

```diff
@@ -15,22 +15,74 @@ def get_chpl_home():
 
 @memoize
 def get_chpl_runtime_incl():
-    default = os.path.join(get_chpl_home(), 'runtime', 'include')
+    prefix = get_install_prefix()
+    if prefix is None:
+        default = os.path.join(get_chpl_home(), 'runtime', 'include')
+    else:
+        chpl_vers = get_chpl_version_from_install()
+        assert chpl_vers is not None
+        default = os.path.join(prefix, 'lib', 'chapel', chpl_vers, 'runtime', 'include')
     chpl_runtime_incl = overrides.get('CHPL_RUNTIME_INCL', default)
     return chpl_runtime_incl
 
 @memoize
 def get_chpl_runtime_lib():
-    default = os.path.join(get_chpl_home(), 'lib')
+    prefix = get_install_prefix()
+    if prefix is None:
+        default = os.path.join(get_chpl_home(), 'lib')
+    else:
+        chpl_vers = get_chpl_version_from_install()
+        assert chpl_vers is not None
+        default = os.path.join(prefix, 'lib', 'chapel', chpl_vers, 'runtime', 'lib')
     chpl_runtime_lib = overrides.get('CHPL_RUNTIME_LIB', default)
     return chpl_runtime_lib
 
 @memoize
 def get_chpl_third_party():
-    default = os.path.join(get_chpl_home(), 'third-party')
+    prefix = get_install_prefix()
+    if prefix is None:
+        default = os.path.join(get_chpl_home(), 'third-party')
+    else:
+        chpl_vers = get_chpl_version_from_install()
+        assert chpl_vers is not None
+        default = os.path.join(prefix, 'lib', 'chapel', chpl_vers, 'third-party')
     chpl_third_party = overrides.get('CHPL_THIRD_PARTY', default)
     return chpl_third_party
 
+install_path_regex = re.compile(
+    '^(.*){}share{}chapel{}(\\d+\\.\\d+)$'.format(os.path.sep,
+                                                  os.path.sep,
+                                                  os.path.sep))
+
+@memoize
+def get_chpl_version_from_install():
+    if get_install_prefix():
+        chpl_home = get_chpl_home()
+        m = install_path_regex.match(chpl_home)
+        # if we are in an installed directory, this should always match
+        assert m is not None
+        return m.group(2)
+    return None
+
+
+@memoize
+def get_install_prefix():
+    """
+    gets the prefix where Chapel is installed, if not installed returns None
+    """
+    chpl_home = get_chpl_home()
+    # if the last three parts of chpl_home are "share", "chapel", and a version number
+    # and there are no makefiles in the chpl_home directory,
+    # then we are in an installed directory
+
+    m = install_path_regex.match(chpl_home)
+    if m is None or os.path.exists(os.path.join(chpl_home, 'Makefile')):
```

Comment:
The `Makefile` will not be installed. I think that there are other ways to detect a non-prefix install. In particular, if the `$CHPL_HOME/bin` and/or `CHPL_HOME/lib` exist, then it's not a prefix install.

I'd recommend changing to decide it's a prefix install if neither `$CHPL_HOME/bin` nor `CHPL_HOME/lib` exist. This should be more robust than the path regex you have.

Reply:
> I'd recommend changing to decide it's a prefix install if neither $CHPL_HOME/bin nor CHPL_HOME/lib exist. 

But that is not strictly true, as a fresh clone of Chapel will not have a `bin` or `lib`. And `printchplenv` can be run in a fresh clone without building anything (and thus creating `bin` and `lib`)

Reply:
That's a good point. Perhaps we need regex + no `$CHPL_HOME/bin` `CHPL_HOME/lib`. Or perhaps there is something else we can use. I will poke at it for a minute.

Reply:
Unpacking a `.tar.gz` source release will lead to some directories that aren't in either installed version:
 * `doc` `examples` `frontend` `highlight` `man` `tools`

What if you detected it like this. Check inside of `$CHPL_HOME`:
 * if there is `doc`, `frontend`, and `tools`, it is not installed
 * otherwise, if there is `bin` and `lib` then it is installed and not a prefix install
 * otherwise, it's a prefix install

I don't object to the regex, by the way. I just think it's good if we can have some more signals here for this code to know it's doing the right thing.

Reply:
(The way the current code looks for `Makefile` seems worrying to me, more than anything)

Reply:
I changed it to use `CHPL_HOME/doc` and `CHPL_HOME/frontend` to detect if its installed, and to check for bin and lib if its a CHPL_HOME install vs prefix install. I left the regex as a final sanity check, partly because its needed to extract the prefix